python/runtime/cudaq/builder/py_kernel_builder.cpp

```diff
@@ -566,7 +566,38 @@ void bindKernel(py::module &mod) {
           "  kernel.apply_call(target_kernel, qubit)\n"
           "  # The final measurement of `qubit` should return the 1-state.\n"
           "  kernel.mz(qubit)\n")
-
+      .def(
+          "for_loop",
+          [](kernel_builder<> &self, std::size_t start, std::size_t end,
+             py::function body) { self.for_loop(start, end, body); },
+          "Add a for loop that starts from the given `start` integer index, "
+          "ends at the given `end` integer index, and applies the given `body` "
```

Comment:
I am interpreting this as the end being inclusive; i.e. for_loop (1,3, body) would have the iterations 1,2, and 3. Is that correct?
Edit: based on the tests below I don't think this range is inclusive, so for_loop (1,3, body) would only have iterations 1 and 2. You could add something like "... ends at the given `end` integer index (non-inclusive), ..."

Reply:
Will update 👍